app/domain/authentication/authn_azure/validate_application_identity.rb

```diff
@@ -0,0 +1,84 @@
+require 'forwardable'
+require 'command_class'
+
+module Authentication
+  module AuthnAzure
+
+    Err = Errors::Authentication::AuthnAzure
+
+    ValidateApplicationIdentity = CommandClass.new(
+        dependencies: {
+            role_class:                ::Role,
+            resource_class:             Resource,
```

Comment:
Has the same meaning?

Reply:
In this case yes. `::` in ruby  is like `/` in bash - it goes to the root module. In case the root module is not  explicitly present then it will look for the closest module that has the class `Role`. 

So in case the module tree is:
```
A -> B -> C
D
```
Then referring to `D` in code of module `B` is the same as referring to `::D`.

But in case the module tree is:
```
A -> B -> C
C
```
Then referring to `C` in code of module `B` is different than referring to `::C` as the first one will go to `A::B::C`.